flow/data_pipeline/query.py

```diff
@@ -88,27 +103,390 @@ class QueryStrings(Enum):
             SELECT
                 id,
                 time_step,
-                target_accel_no_noise_with_failsafe,
+                COALESCE (target_accel_no_noise_with_failsafe,
+                          target_accel_no_noise_no_failsafe,
+                          realized_accel) AS acceleration,
                 road_grade,
                 source_id,
+                speed AS cur_speed,
                 time_step - LAG(time_step, 1)
                   OVER (PARTITION BY id ORDER BY time_step ASC ROWS BETWEEN 1 PRECEDING and CURRENT ROW) AS sim_step,
                 LAG(speed, 1)
                   OVER (PARTITION BY id ORDER BY time_step ASC ROWS BETWEEN 1 PRECEDING and CURRENT ROW) AS prev_speed
-            FROM trajectory_table
+            FROM fact_vehicle_trace
             WHERE 1 = 1
                 AND date = \'{{date}}\'
                 AND partition_name=\'{{partition}}\'
         ), denoised_speed_cte AS (
             SELECT
                 id,
                 time_step,
-                prev_speed + target_accel_no_noise_with_failsafe * sim_step AS speed,
-                target_accel_no_noise_with_failsafe AS acceleration,
+                COALESCE (prev_speed + acceleration * sim_step, cur_speed) AS speed,
+                acceleration,
                 road_grade,
                 source_id
             FROM lagged_timestep
         )
         {}""".format(VEHICLE_POWER_DEMAND_COMBUSTION_FINAL_SELECT.format(1,
                                                                          'POWER_DEMAND_MODEL_DENOISED_ACCEL_VEL',
                                                                          'denoised_speed_cte'))
+
+    FACT_NETWORK_THROUGHPUT_AGG = """
+        WITH min_time AS (
+            SELECT
+                source_id,
+                id,
+                MIN(time_step) AS enter_time
+            FROM fact_vehicle_trace
+            WHERE 1 = 1
+                AND date = \'{date}\'
+                AND partition_name = \'{partition}\'
+                AND x BETWEEN 500 AND 2300
```

Comment:
i think we should probably pull these constants out of the query, so that if anything changes in the future, we only have to change it in one place.
- `control_start_position` (500) or `control_start_edge` ('ghost0')
- `control_end_position` (2300) or `control_end_edge` ('119257908#3')
- `warmup_time` (720)
- `horizon_time` (1200)